nemo/core/neural_modules.py

```diff
@@ -162,6 +169,142 @@ def __is_of_allowed_type(self, var):
+
+            Args:
+                config_file: yml file name
+
+                config_dir: directory where the file will be stored (DEFAULT: ~/data/configs)
+        """
+        # Check if generic export will work.
+        if not self.__validate_params(self._init_params):
+            raise ValueError(
+                "Generic Module export cannot work as some of the values are not primitive types (string, int, float) "
+                F"or (lists of/dicts of) primitive types. Please implement your own custom `export_config()` and "
+                F"`import_config()` methods for your custom Module class."
+            )
+
+        # Greate an absolute path.
+        abs_path_file = path.join(path.expanduser(config_dir), config_file)
+
+        # Create the dictionary to be exported.
+        to_export = {}
+
+        # Get module "full specification".
+        module_full_spec = str(self.__module__) + "." + str(self.__class__.__qualname__)
+        module_class_name = type(self).__name__
+        # print(module_full_spec)
+
+        # Check whether module belongs to a collection.
+        spec_list = module_full_spec.split(".")
+
+        # Do not check Neural Modules from unit tests.
+        if spec_list[0] == "tests":
+            # Set collection variables.
+            collection_type = "tests"
+            collection_version = None
+        else:
+            # Check if component belongs to any collection
+            if len(spec_list) < 3 or (spec_list[0] != "nemo" and spec_list[1] != "collection"):
+                logging.warning(
+                    "Module `{}` does not belong to any collection. This won't be allowed in the next release.".format(
+                        module_class_name
+                    )
+                )
+                collection_type = "unknown"
+                collection_version = None
+            else:
+                # Ok, set collection.
+                collection_type = spec_list[2]
+                collection_version = None
+                # TODO: to be SET!
+                # print(getattr("nemo.collections.nlp", __version__))
+
+        # Add "header" with module "specification".
+        to_export["header"] = {
+            "nemo_core_version": nemo_version,
+            "collection_type": collection_type,
+            "collection_version": collection_version,
+            "class": module_class_name,
+            "full_spec": module_full_spec,
+        }
+        # Add init parameters.
+        to_export["init_params"] = self._init_params
+        # print(to_export)
+
+        # All parameters are ok, let's export.
+        with open(abs_path_file, 'w') as outfile:
+            yaml.dump(to_export, outfile)
+
+        logging.info(
+            "Configuration of module {} ({}) exported to {}".format(self._uuid, type(self).__name__, abs_path_file)
+        )
+
+    @classmethod
+    def import_config(cls, config_file, config_dir="~/data/configs"):
+        """
```

Comment:
+1 for making it a class method.
The name is a bit confusing, imho, shouldn't this be "import_from_config" or just "from_config" or something else? I had to read the docstring to understand that it returns a NM instance

Reply:
Originally in the Design Doc I had import_from_config() and export_to_config(), then I simplified it, but yeah, I think you are right, it is less understandable..

Reply:
done